prow/jenkins/jenkins.go

```diff
@@ -30,18 +30,69 @@ import (
 
 const (
 	maxRetries = 5
-	retryDelay = 4 * time.Second
+	retryDelay = 100 * time.Millisecond
+	buildID    = "buildId"
+)
+
+const (
+	Succeess = "SUCCESS"
+	Failure  = "FAILURE"
+	Aborted  = "ABORTED"
 )
 
 // Status is a build result from Jenkins. If it is still building then
 // Success is meaningless. If it is enqueued then both Success and
 // Number are meaningless.
+// TODO: Remove
 type Status struct {
 	Building bool
 	Success  bool
 	Number   int
 }
 
+type JenkinsBuild struct {
+	Actions []struct {
+		Parameters []struct {
+			Name  string `json:"name"`
+			Value string `json:"value"`
```

Comment:
It seems that if the Jenkins queue contain builds with boolean parameters, it will clobber the queue decoding because I am decoding using strings ([similar to the decoding that happens for running builds](https://github.com/kubernetes/test-infra/blob/c5568d6e054221e3ce8e45f904800b19e92ae95b/prow/jenkins/jenkins.go#L189)). I need to use an interface here and handle the conversion outside of json.Unmarshal.